user/v1/types.go

```diff
@@ -133,3 +133,34 @@ type GroupList struct {
 	// Items is the list of groups
 	Items []Group `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IdentityMetadata represents an instance of identity metadata associated with a single OAuth flow.
+type IdentityMetadata struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// ProviderName is the source of identity information.
+	ProviderName string `json:"providerName" protobuf:"bytes,2,opt,name=providerName"`
+
+	// ProviderGroups is the groups asserted by the provider for this OAuth flow.
+	ProviderGroups []string `json:"providerGroups" protobuf:"bytes,3,opt,name=providerGroups"`
+
+	// ExpiresIn is the seconds from CreationTime before this identityMetadata expires.
+	ExpiresIn int64 `json:"expiresIn,omitempty" protobuf:"varint,4,opt,name=expiresIn"`
```

Comment:
Any reason this shouldn't be `ExpiresAt` with a timestamp? Seems like it would be easier to reason about

Reply:
The convention is to do math based on start time and some interval (see OAuth tokens as example).  I copied this field name exactly from the OAuth token field, but that name is based off of the OAuth spec.  @smarterclayton probably would want me to call this `ExpiresInSeconds` (i.e describe the thing and the units in the name).

Reply:
I agree with what @smarterclayton would probably request because its stated in our api conventions:

https://github.com/kubernetes/community/blob/be9eeca6ee3becfa5b4c96bedf62b5b3ff5b1f8d/contributors/devel/api-conventions.md#units